cluster-autoscaler/clusterstate/clusterstate.go

```diff
@@ -194,6 +195,18 @@ func (csr *ClusterStateRegistry) RegisterOrUpdateScaleUp(nodeGroup cloudprovider
 	csr.registerOrUpdateScaleUpNoLock(nodeGroup, delta, currentTime)
 }
 
+// RegisterScaleUp registers scale-up for give node group or changes requested node increase
+// count.
+func (csr *ClusterStateRegistry) RegisterScaleUp(nodeGroup cloudprovider.NodeGroup, delta int, currentTime time.Time) error {
```

Comment:
I wanted to rename `RegisterOrUpdateScaleUp` but thought otherwise and created another function called `RegisterScaleUp`. 

 I want to use the observer to `RegisterScaleUp` [here](https://github.com/kubernetes/autoscaler/pull/5729/files#diff-75a8e98f58ff9a3dd7cd26373252a59869e6a4e9c30532ac0d30805c27c6ab4fR158) (that's [only place where `RegisterOrUpdateScaleUp` is used](https://github.com/search?q=repo%3Akubernetes%2Fautoscaler%20RegisterOrUpdateScaleUp&type=code)). If we use `RegisterOrUpdateScaleUp`, subscribers to the observer interface will have to implement the same function. I think `RegisterOrUpdateScaleUp` is potentially confusing because the function is exposing too much info about the implementation. User shouldn't have to care about if `RegisterScaleUp` is updating the `scaleUpRequests` struct inside. `RegisterScaleUp` encapsulates the idea of registering the scale-up. Whether that is by creating a new entry in `scaleUpRequests` or by updating an existing one should be an implementation detail (everything else inside `RegisterOrUpdateScaleUp` should be an implementation detail as well). 

Maybe we can rename `RegisterOrUpdateScaleUp` to `RegisterScaleUp` or if you have a better idea, I am open to that as well :)

Reply:
Updating existing function SGTM, my main concern here is we shouldn't have two functions to do the same thing.